openml/datasets/functions.py

```diff
@@ -42,7 +48,8 @@ def _list_cached_datasets():
 
             if "dataset.arff" in dataset_directory_content and \
                     "description.xml" in dataset_directory_content:
-                datasets.append(did)
+                if did not in datasets:
```

Comment:
It could have been there because of line 31, which iterates over two directories. If it was in the 'official' directory, it should not be added again if it is also in the 'private' directory. Because this is not documented and not tested, I opened issue #151 and I will remove this.